azkaban-db/src/main/sql/create.validated_dependencies.sql

```diff
@@ -0,0 +1,7 @@
+CREATE TABLE validated_dependencies (
+  file_name         VARCHAR(100),
+  file_sha1         VARCHAR(40),
+  validation_key    VARCHAR(40),
+  validation_status INT,
+  PRIMARY KEY (file_name, file_sha1, validation_key)
```

Comment:
So here is a thought. It looks like we always query this table by `validation_key` and an arbitrary number of (`file_name`, `file_sha1`) pairs. So if the table is sorted by `file_name` first as implemented, then we'll jump all over the table to fetch all needed rows. Even though each tow will (likely) be accessed by its PK, they are scattered across the table. We'll need to read many table blocks/pages and get (typically) one row from each.
In contrast, if we sorted the table by `validation_key` first, then all relevant rows for each query would always be located sequentially and together, and we'd likely need to read (and keep in buffer cache!) fewer pages, and sequential reads are faster on spinning disks.
Does it make sense? You may want to seek second opinion on this, but if it checks out, please consider changing your PK definition to `(validation_key, file_name, file_sha1)`.